### PR TITLE
ERA A+V upgrade to Avalon 6.5.0 - step 3b of 6: merge upstream v6.5.0 plus Travis fixes

### PR DESCRIPTION
**Pull request by jefferya**
Step 3 of the ERA A+V Avalon update to 6.5.0. Meant to be combined with several other steps to produce a working ERA A+V

This step:
Merge tag 'v6.5.0' of https://github.com/avalonmediasystem/avalon into UAL modified 'v6.4.3' of https://github.com/ualbertalib/avalon
    
Avalon 6.5 includes IIIF manifest generation, a new editor for structural metadata, and integration with the Timeliner tool for annotation of AV materials, as well as a variety of updates for supporting libraries and components.

ualbertalib commits
* fix merge conflicts between ualberta customizations and upstream 6.5.0 code
* update Travis (upstreams appears to have switch to CircleCI)
    
v6.5.0 Upstream added features
    
Structural Metadata Editor:
A visual editor for creating and modifying section structure available on an item's Edit page. The new editor features a waveform visualization and interaction much like audio editing software such as Audacity or Adobe Audition.
    
IIIF Timeliner:
Integration with Timeliner, a reimplementation of the Variations Audio Timeliner annotation tool. Create, edit and share Timelines using media from items within Avalon.
    
IIIF Presentation 3.0 Manifest Generation:
IIIF manifests are now generated by Avalon for all items and can be retrieved from an application endpoint.
    
Updates and Bug Fixes
* Support for Rails 5.2
* Web upload limit on file size has been increased to 2 GB
* Title attribute is encoded properly in HTML embed code (#2930)
* Type-ahead in username input field no longer overrides values of a very short length (#2896)
* Updating collection information no longer removes staff users (#2994)

### CODE REVIEW

File: db/migrate/20190212211753_create_timelines.rb [hunk 1]
```diff
@@ -2,7 +2,7 @@ class CreateTimelines < ActiveRecord::Migration[5.1]
   def change
     create_table :timelines do |t|
       t.string :title
-      t.references :user, foreign_key: true, type: :integer
+      t.references :user, foreign_key: true, type: :bigint
```
Comment (jefferya) [inline db/migrate/20190212211753_create_timelines.rb:5]: Rails 4.x to 5.x changed autogenerated ID columns default type from integer to bigint. I think this was causing Travis failure.  Note: PR #627 (Upgrade step 5) proposes a solution in the context of preexisting data.

File: lib/tasks/avalon.rake [hunk 2]
```diff
@@ -296,9 +296,9 @@ EOC
       Dir[Rails.root.join("spec/factories/**/*.rb")].each {|f| require f}
 
       mf_count = [ENV['master_files'].to_i,1].max
-      mo = FactoryGirl.create(:media_object)
+      mo = FactoryBot.create(:media_object)
```
Comment (jefferya) [inline lib/tasks/avalon.rake:299]: Gem name change.

File: spec/controllers/media_objects_controller_spec.rb [hunk 3]
```diff
@@ -259,8 +259,6 @@
           # master_file_obj = FactoryBot.create(:master_file, master_file.slice(:files))
           media_object = FactoryBot.create(:media_object)#, master_files: [master_file_obj])
           fields = media_object.export_descriptive_metadata_attributes
-          descMetadata_fields.each {|f| fields[f] = media_object.send(f) }
-          # fields = media_object.attributes.select {|k,v| descMetadata_fields.include? k.to_sym }
```
Comment (jefferya) [inline spec/controllers/media_objects_controller_spec.rb:263]: Removed as `media_object.export_descriptive_metadata_attributes`, a locally created function, completes the same function plus more (related to language and PBcore as a datasource). The highlighted is repeated a total of four times in the file

Links to the PR #302 and commit
https://github.com/ualbertalib/avalon/commit/3585bb0212f5032838d4f6d79a13fded9818e900
3585bb0

File: spec/features/secure_routes_spec.rb [hunk 4]
```diff
@@ -49,13 +49,13 @@
...
       visit '/about/health'
-      expect(page).to have_content('Sample Content')
+      expect(page).to have_content('Featured Content')
     end
 
     it "should not provide access to anonymous users" do
       login_as user, scope: :user
       visit '/about/health'
-      expect(page).to have_content('Sample Content')
+      expect(page).to have_content('Featured Content')
```
Comment (jefferya) [inline spec/features/secure_routes_spec.rb:58]: Local change to the splash page, missed this in the upstream code merge. Not certain why the tests for secure routes depend on expectations that the resulting page has a certain content string.
There's a better way in Ruby, right? I'll search Jupiter when I have time.

File: spec/models/master_file_spec.rb [hunk 5]
```diff
@@ -563,8 +563,12 @@ class MyEncoder < ActiveEncode::Base
 
   describe "waveform generation" do
     subject(:master_file) { FactoryBot.create(:master_file) }
+    let(:collection) { FactoryBot.create(:collection, name: 'rails env waveform test') }
+    let(:media_object) { MediaObject.new }
 
     it 'runs the waveform job' do
+      media_object.collection = collection
+      master_file.media_object = media_object
       expect(WaveformJob).to receive(:perform_now).with(master_file.id)
```
Comment (jefferya) [inline spec/models/master_file_spec.rb:572]: Fixes a test failure depending the config setting, SETTINGS__MASTER_FILE_MANAGEMENT__STRATEGY, that moves file uploaded via the UI into a useful location/name and not leave in the `upload` directory with the potential to be overwritten with a file of the same name https://github.com/ualbertalib/avalon/pull/454

File: spec/integration/working_file_path_spec.rb [hunk 6]
```diff
@@ -198,7 +198,25 @@
...
+        example.run
+
+        Rails.application.secrets.matterhorn_client_media_path = old_media_path
+      ensure
+        FileUtils.remove_entry media_path
+      end
+    end
+
     it 'returns blank' do
+      expect(Rails.application.secrets.matterhorn_client_media_path).to be_blank
```
Comment (jefferya) [inline spec/integration/working_file_path_spec.rb:219]: Testing against local settings, the above code remove upstream assumption that the media path is unset. Purpose, allow tests to work against settings used locally.

**Comment by jefferya** [comment]
@mbarnett 
Added annotations for the changes. Ready for review.

File: spec/features/secure_routes_spec.rb (hunk 4 above)
Comment (mbarnett) [reply spec/features/secure_routes_spec.rb:58]: I can think a couple of better ways to do that, yeah, depending on what the implementation is like. Usually what you'd do is redirect the unauthorized user to a message or login page or whatever when they visit the secure URL, and then in the test just assert that you got redirected to where you expected them to be sent.

**Pull request merged**